site/src/lib/s3.ts

```diff
@@ -0,0 +1,71 @@
+import { S3Client } from "@aws-sdk/client-s3";
+import * as envalid from "envalid";
+
+let s3Client: S3Client | undefined;
+
+export const getS3Client = (): S3Client => {
+  if (!s3Client) {
+    const env = envalid.cleanEnv(process.env, {
+      S3_API_ENDPOINT: envalid.str(),
+      S3_BUCKET: envalid.str(),
+      S3_ACCESS_KEY_ID: envalid.str(),
+      S3_SECRET_ACCESS_KEY: envalid.str(),
+    });
+
+    s3Client = new S3Client({
+      endpoint: env.S3_API_ENDPOINT,
+      region: "us-east-1",
+      credentials: {
+        accessKeyId: env.S3_ACCESS_KEY_ID,
+        secretAccessKey: env.S3_SECRET_ACCESS_KEY,
+      },
+      forcePathStyle: true,
+    });
+  }
+
+  return s3Client;
+};
+
+let s3Bucket: string | undefined;
+
+export const getS3Bucket = (): string => {
+  if (!s3Bucket) {
+    const env = envalid.cleanEnv(process.env, {
+      S3_BUCKET: envalid.str(),
+    });
+
+    s3Bucket = env.S3_BUCKET;
+  }
+
+  return s3Bucket;
+};
+
+export const resolveS3ResourcePath = (
+  category: "blocks" | "avatars",
+  subpath: string,
+): string => {
+  return `/${category}/${subpath}`;
```

Comment:
The leading slash is intepreted as a directory with no name, which isn't handled properly / makes assets 404

```suggestion
  return `${category}/${subpath}`;
```

raw keys (I uploaded `1.txt` directly)
<img width="728" alt="Screenshot 2022-10-04 at 15 56 52" src="https://user-images.githubusercontent.com/37743469/193853505-925fbc3a-9d10-4696-9008-43f8cb9e0a63.png">

inferred folder structure
<img width="775" alt="Screenshot 2022-10-04 at 15 54 43" src="https://user-images.githubusercontent.com/37743469/193853539-a2ed5f46-5488-4515-9aa3-828cec0c161a.png">